cmd/ttn-lw-cli/commands/notifications.go

```diff
@@ -0,0 +1,144 @@
+// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"os"
+
+	"github.com/TheThingsIndustries/protoc-gen-go-flags/flagsplugin"
+	"github.com/spf13/cobra"
+	"go.thethings.network/lorawan-stack/v3/cmd/internal/io"
+	"go.thethings.network/lorawan-stack/v3/cmd/ttn-lw-cli/internal/api"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+)
+
+var notificationsCommand = &cobra.Command{
+	Use:     "notifications",
+	Aliases: []string{"notification"},
+	Short:   "Manage notifications",
+}
+
+var notificationsListCommand = &cobra.Command{
+	Use:   "list",
+	Short: "List notifications",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		req := &ttnpb.ListNotificationsRequest{
+			Status: []ttnpb.NotificationStatus{
+				ttnpb.NotificationStatus_NOTIFICATION_STATUS_UNSEEN,
+			},
+		}
+		_, err := req.SetFromFlags(cmd.Flags(), "")
+		if err != nil {
+			return err
+		}
+		if len(args) > 0 && req.GetReceiverIds().GetUserId() == "" {
+			if len(args) > 1 {
+				logger.Warn("Multiple IDs found in arguments, considering only the first")
+			}
+			req.ReceiverIds = &ttnpb.UserIdentifiers{UserId: args[0]}
+		}
+
+		_, _, opt, getTotal := withPagination(cmd.Flags())
+
+		is, err := api.Dial(ctx, config.IdentityServerGRPCAddress)
+		if err != nil {
+			return err
+		}
+		svc := ttnpb.NewNotificationServiceClient(is)
+
+		res, err := svc.List(ctx, req, opt)
+		if err != nil {
+			return err
+		}
+		getTotal()
+
+		if err = io.Write(os.Stdout, config.OutputFormat, res.Notifications); err != nil {
+			return err
+		}
+
+		if markAsSeen, _ := cmd.Flags().GetBool("mark-as-seen"); !markAsSeen && len(res.Notifications) > 0 {
```

Comment:
I miss why the logic is reversed here. So you're marking them as unseen if the flag is false, but where are they marked as seen ? I see that the store implementation uses the read only store, so `List` cannot have side effects.

Reply:
Oh yeah this is completely wrong. Don't know what I was thinking.